shared/js/background/classes/agentspoofer.es6.js

```diff
@@ -0,0 +1,145 @@
+const agentStorage = require('./../storage/agents.es6')
+const agentparser = require('useragent')
+const trackers = require('../trackers.es6')
+const tldts = require('tldts')
+const tabManager = require('../tab-manager.es6')
+
+class AgentSpoofer {
+    constructor () {
+        this.realAgent = navigator.userAgent
+        this.parsedAgent = agentparser.lookup(this.realAgent)
+        this.spoofedAgent = this.realAgent
+        this.selectAgent()
+        this.needsRotation = true
+        this.tabs = {} // maintain cache of tab root URL's (chrome only)
```

Comment:
Is that something that we should merge into tabManager? 🤔

Reply:
I should have been using tabs from the start for this - much simpler.